src/execution/subscribe.ts

```diff
@@ -220,22 +261,32 @@ async function executeSubscription(
     // Call the `subscribe()` resolver or the default resolver to produce an
     // AsyncIterable yielding raw payloads.
     const resolveFn = fieldDef.subscribe ?? exeContext.subscribeFieldResolver;
-    const eventStream = await resolveFn(rootValue, args, contextValue, info);
+    const eventStream = resolveFn(rootValue, args, contextValue, info);
 
-    if (eventStream instanceof Error) {
-      throw eventStream;
+    if (isPromise(eventStream)) {
+      return eventStream.then(assertEventStream).then(undefined, (error) => {
```

Comment:
It's a direct match on how this is done in `execute` just in one line:
https://github.com/graphql/graphql-js/blob/ea1894ae22bd732f1dc7fa4f7bf21f1525688e9a/src/execution/execute.ts#L516-L518

https://github.com/graphql/graphql-js/blob/ea1894ae22bd732f1dc7fa4f7bf21f1525688e9a/src/execution/execute.ts#L533-L536

Reply:
There additional cost of micro queue trip only if the stuff in `assertEventStream` throws.
Otherwise, nothing changed.